docs/proposals/decentralized-identity-management.md

```diff
@@ -0,0 +1,177 @@
+
+<img src="../images/security/SpireEdgev0.1-CreateCloudNodeSequence.png">
+
+For edge node registration, 
+
++ Edge node entry is created at kubernetes api server using kubectl command. 
+
++ Based on the node type, master or slave, respective edge component are assigned a spiffe id with attestation information as configured.
+
++ Identity controller creates a new entry with spire server.
+
++ Cloud spire agent performs node attestation and receives node certificates. The node certificates are used to connect to spire server through node api for workload certificate requests.
+
++ Edge hub implements spire workload api to retrieve certificates. Cloud spire agent (at edge node) performs workload attestation. On successful attestation, certificate signing request is made to cloud spire server. On response from cloud spire server, cloud spire agent caches the certificate and responds to cloud hub with private key and certificate.
+
++ Edge spire server initialization can be performed with two options. 
+1) (Recommended) Edge spire server can download workload api certificates and connect to edgehub (or metamanager?) to download intermediate CA certificates.
+2) Edge spire server can download workload api certificates and connect to cloud spire server directly to download intermediate CA certificates.
+
++ EdgeD on initialization creates node entry with edge spire server and updates the status to kubernetes api server through hub interface.
+
+<img src="../images/security/SpireEdgev0.1-CreateEdgeNodeSequence.png">
+
+To delete node,
+
+<img src="../images/security/SpireEdgev0.1-DeleteNodeSequence.png">
+
+### Application registration and deletion
+
+On addition of a new user application pod, edged creates entry at edge node spire server, which enables application to retrieve and rotate certificates.
+
+<img src="../images/security/SpireEdgev0.1-CreatePodSequence.png">
+
+For delete pod,
+
+<img src="../images/security/SpireEdgev0.1-DeletePodSequence.png">
+
+## Offline scenarios
+When there is intermittent / no connectivity between the edge node and the cloud , edge applications can still rotate certificates based on TTL configured for edge spire server.
+
+## Scalability
+In the current design, the solution is scalable for single cluster of edge application deployment under a node. Applications hosted under different edge spire server requires a federated spire server deployment for east-to-west direct communication between applications. Also, a policy management framework is necessary for controlling the authorization between applications.
+
+## Open questions
+- How are edge nodes identified from other kubernetes nodes deployed in a cluster (heterogenous deployments) ?
```

Comment:
Yes, this would be important if we support edge to edge direct communication. 
For now, we'll route the data flow from one node to cloud; and then to the other. So the upper logic of node attestation done at cloud spire server is still valid.